### PR TITLE
Voa 541

### PR DESCRIPTION
**Pull request by Painyjames**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -24,3 +24,5 @@ govuk_modules
 npm-debug.log
 *.DS_Store
 frontend/sass/elements
+package-lock.json
```
Comment (rhubner) [inline .gitignore]: https://docs.npmjs.com/files/package-lock.json

File: app/controllers/SaveForLater.scala [hunk 2]
```diff
@@ -97,6 +97,10 @@ object SaveForLater extends FrontendController {
     Redirect(routes.LoginController.show()).withNewSession
   }
 
+  def timeout = Action.async(parse.empty) { implicit request =>
```
Comment (rhubner) [inline app/controllers/SaveForLater.scala]: Doesn't need to be async.

File: conf/application.conf [hunk 3]
```diff
@@ -89,13 +95,14 @@ agentApi {
...
 }
-  
+
 analytics {
   ga {
     trackingCode: "ADDYOURS"
     startDate: "ADDYOURS"
     returnCount: "ADDYOURS"
     refNum: "ADDYOURS"
+    GMT: "ADDYOURS"
```
Comment (rhubner) [inline conf/application.conf]: GTM - Google Tag Manager

File: app/views/layouts/govuk_template2.scala.html [hunk 4]
```diff
@@ -85,6 +101,15 @@
   </head>
 
   <body class="@bodyClasses.getOrElse("")">
+    <!-- Google Tag Manager (noscript) -->
+    <noscript><iframe src="https://www.googletagmanager.com/ns.html?id=GTM-@{current.configuration.getString("analytics.ga.GMT").get}"
+                      height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
+    <!-- End Google Tag Manager (noscript) -->
+    @if(needsTimeoutWarning){
```
Comment (rhubner) [inline app/views/layouts/govuk_template2.scala.html]: Why we just don't embed dome javascript code instant of form inputs?

Comment (Painyjames) [reply app/views/layouts/govuk_template2.scala.html]: that's pretty much what Andrew told me to paste, so I rather do as he says

File: .gitignore (hunk 1 above)
Comment (Painyjames) [reply .gitignore]: yeah, I know that. It's a generated file which usually we shouldn't commit but yes, committing this one will solve the "but it works on my machine" issues.

File: app/controllers/SaveForLater.scala (hunk 2 above)
Comment (Painyjames) [reply app/controllers/SaveForLater.scala]: force of habit

File: conf/application.conf (hunk 3 above)
Comment (Painyjames) [reply conf/application.conf]: yup, changing that

File: app/views/layouts/govuk_template2.scala.html (hunk 4 above)
Comment (Painyjames) [reply app/views/layouts/govuk_template2.scala.html]: https://jira.tools.tax.service.gov.uk/browse/VOA-121
>Copy the following snippet and paste it immediately after the opening <body> tag on every page of RALD, Contact Finder, Digital Contact service, ROFR and BARS.

**Pull request merged**